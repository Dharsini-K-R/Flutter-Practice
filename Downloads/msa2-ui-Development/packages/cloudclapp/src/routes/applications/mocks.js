export const dockerImage = [
  {
    id: "",
    name: "node",
    slug: "node",
    type: "image",
    publisher: {
      id: "docker",
      name: "Docker",
    },
    created_at: "2016-06-17T17:45:46.898676Z",
    updated_at: "2022-02-04T06:12:22.541527Z",
    short_description:
      "Node.js is a JavaScript-based platform for server-side and networking applications.",
    source: "store",
    popularity: 0,
    categories: [
      {
        name: "application_infrastructure",
        label: "Application Infrastructure",
      },
    ],
    operating_systems: [
      {
        name: "linux",
        label: "Linux",
      },
    ],
    architectures: [
      {
        name: "arm64",
        label: "ARM 64",
      },
      {
        name: "s390x",
        label: "IBM Z",
      },
      {
        name: "ppc64le",
        label: "PowerPC 64 LE",
      },
      {
        name: "386",
        label: "386",
      },
      {
        name: "arm",
        label: "ARM",
      },
      {
        name: "amd64",
        label: "x86-64",
      },
    ],
    logo_url: {
      large:
        "https://d1q6f0aelx0por.cloudfront.net/product-logos/library-node-logo.png",
      small:
        "https://d1q6f0aelx0por.cloudfront.net/product-logos/library-node-logo.png",
    },
    certification_status: "not_certified",
    star_count: 11111,
    pull_count: "1B+",
    filter_type: "official",
  },
];

export const VMImage = [
  {
    architecture: "x86_64",
    creationDate: "2022-04-28T21:51:16.000Z",
    deprecationTime: "2024-04-28T21:51:16.000Z",
    description: "Amazon Linux 2 Kernel 5.10 AMI 2.0.20220426.0 x86_64 HVM gp2",
    enaSupport: true,
    hypervisor: "xen",
    imageId: "ami-0022f774911c1d690",
    imageLocation: "amazon/amzn2-ami-kernel-5.10-hvm-2.0.20220426.0-x86_64-gp2",
    imageOwnerAlias: "amazon",
    imageType: "machine",
    isPublic: false,
    name: "amzn2-ami-kernel-5.10-hvm-2.0.20220426.0-x86_64-gp2",
    ownerId: "137112412989",
    platform: "Amazon Linux",
    productCodes: [],
    rootDeviceName: "/dev/xvda",
    rootDeviceType: "ebs",
    virtualizationType: "hvm",
  },
  {
    architecture: "x86_64",
    creationDate: "2022-04-28T21:51:16.000Z",
    deprecationTime: "2024-04-28T21:51:16.000Z",
    description: "Amazon Linux 2 Kernel 5.10 AMI 2.0.20220426.0 x86_64 HVM gp2",
    enaSupport: true,
    hypervisor: "xen",
    imageId: "ami-0022f774911c1d690",
    imageLocation: "amazon/amzn2-ami-kernel-5.10-hvm-2.0.20220426.0-x86_64-gp2",
    imageOwnerAlias: "amazon",
    imageType: "machine",
    isPublic: false,
    name: "amzn2-ami-kernel-5.10-hvm-2.0.20220426.0-x86_64-gp2",
    ownerId: "137112412989",
    platform: "Amazon Linux",
    productCodes: [],
    rootDeviceName: "/dev/xvda",
    rootDeviceType: "ebs",
    virtualizationType: "hvm",
  },
  {
    architecture: "x86_64",
    creationDate: "2022-04-28T21:51:16.000Z",
    deprecationTime: "2024-04-28T21:51:16.000Z",
    description: "Amazon Linux 2 Kernel 5.10 AMI 2.0.20220426.0 x86_64 HVM gp2",
    enaSupport: true,
    hypervisor: "xen",
    imageId: "ami-0022f774911c1d690",
    imageLocation: "amazon/amzn2-ami-kernel-5.10-hvm-2.0.20220426.0-x86_64-gp2",
    imageOwnerAlias: "amazon",
    imageType: "machine",
    isPublic: false,
    name: "amzn2-ami-kernel-5.10-hvm-2.0.20220426.0-x86_64-gp2",
    ownerId: "137112412989",
    platform: "Amazon Linux",
    productCodes: [],
    rootDeviceName: "/dev/xvda",
    rootDeviceType: "ebs",
    virtualizationType: "hvm",
  },
  {
    architecture: "x86_64",
    creationDate: "2022-04-28T21:51:16.000Z",
    deprecationTime: "2024-04-28T21:51:16.000Z",
    description: "Amazon Linux 2 Kernel 5.10 AMI 2.0.20220426.0 x86_64 HVM gp2",
    enaSupport: true,
    hypervisor: "xen",
    imageId: "ami-0022f774911c1d690",
    imageLocation: "amazon/amzn2-ami-kernel-5.10-hvm-2.0.20220426.0-x86_64-gp2",
    imageOwnerAlias: "amazon",
    imageType: "machine",
    isPublic: false,
    name: "amzn2-ami-kernel-5.10-hvm-2.0.20220426.0-x86_64-gp2",
    ownerId: "137112412989",
    platform: "Amazon Linux",
    productCodes: [],
    rootDeviceName: "/dev/xvda",
    rootDeviceType: "ebs",
    virtualizationType: "hvm",
  },
  {
    architecture: "x86_64",
    creationDate: "2022-04-28T21:51:16.000Z",
    deprecationTime: "2024-04-28T21:51:16.000Z",
    description: "Amazon Linux 2 Kernel 5.10 AMI 2.0.20220426.0 x86_64 HVM gp2",
    enaSupport: true,
    hypervisor: "xen",
    imageId: "ami-0022f774911c1d690",
    imageLocation: "amazon/amzn2-ami-kernel-5.10-hvm-2.0.20220426.0-x86_64-gp2",
    imageOwnerAlias: "amazon",
    imageType: "machine",
    isPublic: false,
    name: "amzn2-ami-kernel-5.10-hvm-2.0.20220426.0-x86_64-gp2",
    ownerId: "137112412989",
    platform: "Amazon Linux",
    productCodes: [],
    rootDeviceName: "/dev/xvda",
    rootDeviceType: "ebs",
    virtualizationType: "hvm",
  },
];
