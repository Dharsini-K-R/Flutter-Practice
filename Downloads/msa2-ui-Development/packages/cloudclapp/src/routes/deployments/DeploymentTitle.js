import React from "react";
import { useSelector, useDispatch } from "react-redux";
import { Grid, Typography } from "@material-ui/core";
import { makeStyles, Tooltip } from "@material-ui/core";
import {
  fetchEnvironments,
  getCloudVendors,
} from "cloudclapp/src/store/designations";
import CloudVendor, {
  getServiceFromEnvironment,
} from "cloudclapp/src/services/CloudVendor";
import StatusBadgeIcon from "cloudclapp/src/components/status-badge-icon/StatusBadgeIcon";
import AddItemsButton from "cloudclapp/src/components/add-items-button/AddItemsButton";
import AddDeploymentDialog from "./AddDeploymentDialog";
import { useTranslation } from "react-i18next";
import { getPermission } from "cloudclapp/src/store/designations";

const useStyles = makeStyles(({ palette }) => {
  return {
    boxWrapper: {
      width: "100%",
      borderBottom: "1px solid #e0e2e7",
    },
    icon: {
      fontSize: "31px",
      color: palette.common.black,
    },
    text: {
      fontSize: "19px",
      fontWeight: 600,
      color: palette.common.black,
      marginLeft: 10,
    },
    addIcon: {
      fontSize: 18,
      color: palette.primary.main,
      verticalAlign: "middle",
    },
    buttonText: {
      fontSize: 16,
      color: palette.primary.main,
      marginLeft: 6,
    },
    boxPadding: {
      padding: "0 20px 14px 12px",
    },
    titleContainer: {
      display: "flex",
      padding: "6px 8px",
      "&:hover": {
        backgroundColor: "rgba(56, 64, 82, 0.04)",
      },
    },
  };
});
const DeploymentTitle = ({ id, title, status, environment }) => {
  const classes = useStyles();
  const dispatch = useDispatch();
  const { t } = useTranslation();
  const cloudVendors = useSelector(getCloudVendors);
  const cloudService = getServiceFromEnvironment(environment, cloudVendors);
  const { appWFUri } = CloudVendor.getWFUrisFromEnvironment(
    environment,
    cloudVendors,
  );
  const [showAddDeployment, setShowAddDeployment] = React.useState(false);

  const canCreateDeployment = useSelector(
    getPermission("deployments", "general", "create"),
  );

  const disableAddDeployment =
    environment?.serviceId === null || environment?.envEntityId === 0;

  const openAddDeploymentDialog = () => {
    setShowAddDeployment(true);
  };
  const closeAddDeploymentDialog = () => {
    dispatch(fetchEnvironments);
    setShowAddDeployment(false);
  };
  return (
    <div className={classes.boxWrapper}>
      <Grid container className={classes.boxPadding}>
        <Grid container item xs>
          <div className={classes.titleContainer}>
            <Grid item>
              <StatusBadgeIcon status={status} />
            </Grid>
            <Grid item>
              <Typography
                id={`DEPLOYMENTS_DEPLOYMENT_TITLE_${id}`}
                className={classes.text}
              >
                {title}
              </Typography>
            </Grid>
          </div>
        </Grid>
        {canCreateDeployment && (
          <Tooltip
            title={
              disableAddDeployment
                ? t("You cannot add Deployment on this Environment")
                : ""
            }
          >
            <Grid
              container
              item
              xs={6}
              justifyContent="flex-end"
              style={{ alignItems: "center" }}
            >
              <AddItemsButton
                id={`DEPLOYMENTS_ADD_DEPLOYMENT_BUTTON_${id}`}
                onClickCallBack={openAddDeploymentDialog}
                buttonLabel={t("Add Deployment")}
                disabled={disableAddDeployment}
              ></AddItemsButton>
            </Grid>
          </Tooltip>
        )}
      </Grid>
      {showAddDeployment && (
        <AddDeploymentDialog
          onClose={closeAddDeploymentDialog}
          ubiqubeId={environment.envUbiqubeId}
          workflowPath={appWFUri}
          entityId={environment.envEntityId}
          showOnlyVMs={
            cloudService?.imageType === "vm" ? "virtual-machine" : "docker-hub"
          }
        />
      )}
    </div>
  );
};

export default DeploymentTitle;
