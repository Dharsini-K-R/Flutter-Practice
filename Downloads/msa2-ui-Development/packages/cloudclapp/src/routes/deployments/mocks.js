import { DEPLOYMENT_VARIABLES_NAME } from "cloudclapp/src/Constants";

export const environments = [
  {
    envName: "Development",
    envId: 99,
    envReference: "68cee9b1-9226-4fdd-a4be-0e1e57977139",
    status: "OK",
    cclapOwner: 0,
    deployments: [
      {
        deploymentId: "126016",
        deploymentName: "My_Big_Shop_Test_4",
        deploymentUser: "test-user",
        status: "RUNNING",
      },
      {
        deploymentId: "126088",
        deploymentName: "My_Big_Shop_Test_4",
        deploymentUser: "test-user",
        status: "FAIL",
      },
      {
        deploymentId: "126018",
        deploymentName: "My_Big_Shop_Test_4",
        deploymentUser: "test-user",
        status: "ENDED",
      },
    ],
    envEntityId: 288,
  },
  {
    envName: "QA",
    envId: 118,
    envReference: "fd8016e0-d367-4179-956e-3f5f35a8d764",
    status: "NEVERREACHED",
    cclapOwner: 105,
    deployments: [
      {
        deploymentId: "126019",
        deploymentName: "Dev_My_Big_Shop_Test_4",
        deploymentUser: "test-user",
        status: "RUNNING",
      },
      {
        deploymentId: "126020",
        deploymentName: "My_Big_Shop_Test_4",
        deploymentUser: "test-user",
        status: "FAIL",
      },
      {
        deploymentId: "126021",
        deploymentName: "My_Big_Shop_Test_4",
        deploymentUser: "test-user",
        status: "ENDED",
      },
    ],
    envEntityId: 289,
  },
  {
    envName: "Staging",
    envId: 119,
    envReference: "fd8016e0-d367-4179-956e-3f5f35a8d764",
    status: "NEVERREACHED",
    cclapOwner: 105,
    deployments: [
      {
        deploymentId: "126022",
        deploymentName: "My_Big_Shop_Test_4",
        deploymentUser: "test-user",
        status: "RUNNING",
      },
      {
        deploymentId: "126023",
        deploymentName: "My_Big_Shop_Test_4",
        deploymentUser: "test-user",
        status: "FAIL",
      },
      {
        deploymentId: "126024",
        deploymentName: "My_Big_Shop_Test_4",
        deploymentUser: "test-user",
        status: "ENDED",
      },
    ],
    envEntityId: 289,
  },
];

export const applications = {
  [DEPLOYMENT_VARIABLES_NAME.APPLICATION_NAME]: "NATS0",
  [DEPLOYMENT_VARIABLES_NAME.APPLICATION_SLUG]: "NATS0",
  [DEPLOYMENT_VARIABLES_NAME.APPLICATION_VERSION]: "V2.1.9",
  [DEPLOYMENT_VARIABLES_NAME.APPLICATION_DESCRIPTION]:
    "Amazon Web Services provides Information Technology infrastructure services to businesses in the form of web services",
  [DEPLOYMENT_VARIABLES_NAME.APPLICATION_LOGO]:
    "https://d1q6f0aelx0por.cloudfront.net/product-logos/library-nats-logo.png",
};

export const environment = {
  envName: "hto env 24",
  envId: 126,
  envUbiqubeId: "005A126",
  description: "aslkd adsg aslgd ll",
  cloudVendor: "aws",
  cloudService: "eks",
  orgName: "hto Org extRef",
  envIp: "127.0.0.1",
  envHostname: "temp-host",
  username: "a",
  envReference: "88faa4e6-be22-444f-a085-e362273d76af",
  importFlag: false,
  serviceId: "123",
  serviceName: "Env",
  envWFUri:
    "Process/cloudclapp-wf/Provision_an_EKS_Cluster__AWS_/Provision_an_EKS_Cluster__AWS_",
  appWFUri:
    "Process/cloudclapp-wf/Provision_an_EKS_Apps__AWS_/Provision_an_EKS_Apps__AWS_",
  deploymentCount: 1,
  userCount: 1,
  scanData: null,
  cclapOwner: 105,
  envEntityId: 306,
  status: "Completed",
};
