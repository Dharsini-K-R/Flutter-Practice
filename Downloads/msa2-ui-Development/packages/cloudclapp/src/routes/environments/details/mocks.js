import { DEPLOYMENT_VARIABLES_NAME } from "cloudclapp/src/Constants";
export const deployments = [
  {
    envId: 126,
    deployments: [
      {
        deploymentId: "126027",
        deploymentName: "New APP deployment",
      },
      {
        deploymentId: "126016",
        deploymentName: "New APP deployment 3",
      },
    ],
  },
  {
    envId: 118,
    deployments: [
      {
        deploymentId: "126029",
        deploymentName: "New APP deployment 2",
      },
    ],
  },
];

export const environment = {
  envName: "hto env 24",
  envId: 126,
  envUbiqubeId: "005A126",
  description: "aslkd adsg aslgd ll",
  cloudVendor: "aws",
  cloudService: "eks",
  orgName: "hto Org extRef",
  envIp: "127.0.0.1",
  envHostname: "temp-host",
  username: "a",
  envReference: "88faa4e6-be22-444f-a085-e362273d76af",
  importFlag: false,
  serviceId: null,
  serviceName: "Env",
  envWFUri:
    "Process/cloudclapp-wf/Provision_an_EKS_Cluster__AWS_/Provision_an_EKS_Cluster__AWS_",
  appWFUri:
    "Process/cloudclapp-wf/Provision_an_EKS_Apps__AWS_/Provision_an_EKS_Apps__AWS_",
  deploymentCount: 1,
  userCount: 1,
  scanData: null,
  cclapOwner: 105,
  envEntityId: 306,
  status: "Completed",
};

export const applications = [
  {
    [DEPLOYMENT_VARIABLES_NAME.APPLICATION_NAME]: "NATS0",
    [DEPLOYMENT_VARIABLES_NAME.APPLICATION_SLUG]: "NATS0",
    [DEPLOYMENT_VARIABLES_NAME.APPLICATION_VERSION]: "V2.1.9",
    [DEPLOYMENT_VARIABLES_NAME.APPLICATION_DESCRIPTION]:
      "Amazon Web Services provides Information Technology infrastructure services to businesses in the form of web services",
    [DEPLOYMENT_VARIABLES_NAME.APPLICATION_LOGO]:
      "https://d1q6f0aelx0por.cloudfront.net/product-logos/library-nats-logo.png",
  },
];

export const cloudVendor = { vendor: "aws", service: "eks" };

export const workflowData = [
  {
    name:
      "Process/cloudclapp-wf/Provision_Apps_EKS_Cluster/Process_Generate_Application_WF",
    displayName: "Restart APP",
    type: "UPDATE",
    visibility: 5,
    allowSchedule: null,
    tasks: null,
  },
  {
    name: "Process/cloudclapp-wf/Provision_Apps_EKS_Cluster/Process_Resume_APP",
    displayName: "Resume APP",
    type: "UPDATE",
    visibility: 5,
    allowSchedule: null,
    tasks: null,
  },
  {
    name: "Process/cloudclapp-wf/Provision_Apps_EKS_Cluster/Process_Pause_APP",
    displayName: "Pause APP",
    type: "UPDATE",
    visibility: 5,
    allowSchedule: null,
    tasks: null,
  },
];

export const userList = [
  {
    id: 130,
    baseRole: {
      id: 2,
      name: "Administrator",
    },
    externalReference: "NCLG130",
    login: "hto@openmsa.co",
    operator: {
      id: 33,
      abonnes: [],
      baseUrl: null,
      isPartner: false,
      name: "hto org 1103",
      prefix: "00u",
      address: {
        city: "",
        country: "",
        fax: "",
        mail: "",
        phone: "",
        streetName1: "",
        streetName2: "",
        streetName3: "",
        zipCode: "",
      },
      customersCount: 0,
      adminsCount: 0,
      managersCount: 0,
      externalReference: "hto org 1103",
    },
    acteurId: 495,
    address: {
      city: "",
      country: "",
      fax: null,
      mail: "hto@openmsa.co",
      phone: "",
      streetName1: "",
      streetName2: "",
      streetName3: "",
      zipCode: "",
    },
    delegationProfileId: 0,
    userType: 2,
    delegations: null,
    delegationsPerCustomer: {},
    netceloId: {
      ubiID: "00uG130",
      thePrefix: "00u",
      id: 130,
      customerPrefix: "00u",
    },
    isExternalAuth: false,
    activationKey: null,
    ccla: false,
    attachedCustomerIds: null,
    firstname: "",
    attachedOperatorIds: [33],
    manageAllUsers: false,
    name: "hto@openmsa.co",
    sort: "NAME",
    ldapAuthentication: false,
    delegationProfilePerCustomer: {},
  },
  {
    id: 125,
    baseRole: {
      id: 2,
      name: "Administrator",
    },
    externalReference: "NCLG125",
    login: "hto@openmsa.com",
    operator: {
      id: 29,
      abonnes: [],
      baseUrl: null,
      isPartner: false,
      name: "brand new hto org!",
      prefix: "00p",
      address: {
        city: "",
        country: "",
        fax: "",
        mail: "",
        phone: "",
        streetName1: "",
        streetName2: "",
        streetName3: "",
        zipCode: "",
      },
      customersCount: 0,
      adminsCount: 0,
      managersCount: 0,
      externalReference: "htoExtRef",
    },
    acteurId: 454,
    address: {
      city: "",
      country: "",
      fax: null,
      mail: "hto@openmsa.com",
      phone: "",
      streetName1: "",
      streetName2: "",
      streetName3: "",
      zipCode: "",
    },
    delegationProfileId: 0,
    userType: 2,
    delegations: null,
    delegationsPerCustomer: {},
    netceloId: {
      ubiID: "00pG125",
      thePrefix: "00p",
      id: 125,
      customerPrefix: "00p",
    },
    isExternalAuth: false,
    activationKey: null,
    ccla: false,
    attachedCustomerIds: null,
    firstname: "",
    attachedOperatorIds: [29],
    manageAllUsers: false,
    name: "hto@openmsa.com",
    sort: "NAME",
    ldapAuthentication: false,
    delegationProfilePerCustomer: {},
  },
];

export const monitoringGraph = [
  {
    id: "AVAILABILITY",
    name: "Availability",
    kpis: {
      latency: {
        color: "#24a36a",
        label: "Latency",
        id: "latency",
        suffix: "ms",
      },
      ttl: {
        color: "#0059a7",
        label: "TTL",
        id: "ttl",
      },
    },
    verticalLabel: "y",
    data: [
      {
        time: 1648485360000,
        latency: 0,
        ttl: 64,
      },
      {
        time: 1648485600000,
        latency: 0,
        ttl: 64,
      },
      {
        time: 1648485840000,
        latency: 0.25,
        ttl: 64,
      },
      {
        time: 1648486080000,
        latency: 0,
        ttl: 64,
      },
      {
        time: 1648486320000,
        latency: 0,
        ttl: 64,
      },
      {
        time: 1648486560000,
        latency: 2500,
        ttl: 64,
      },
      {
        time: 1648486800000,
        latency: 0,
        ttl: 64,
      },
      {
        time: 1648487040000,
        latency: 0,
        ttl: 64,
      },
      {
        time: 1648487280000,
        latency: 0,
        ttl: 64,
      },
      {
        time: 1648487520000,
        latency: 0,
        ttl: 64,
      },
      {
        time: 1648487760000,
        latency: 0,
        ttl: 64,
      },
      {
        time: 1648488000000,
        latency: 0,
        ttl: 64,
      },
      {
        time: 1648488240000,
        latency: 0,
        ttl: 64,
      },
      {
        time: 1648488480000,
        latency: 0,
        ttl: 64,
      },
      {
        time: 1648488720000,
        latency: 0,
        ttl: 64,
      },
      {
        time: 1648488960000,
        latency: 0,
        ttl: 64,
      },
      {
        time: 1648489200000,
        latency: 0,
        ttl: 64,
      },
      {
        time: 1648489440000,
        latency: 0,
        ttl: 64,
      },
      {
        time: 1648489680000,
        latency: 0,
        ttl: 64,
      },
      {
        time: 1648489920000,
        latency: 0,
        ttl: 64,
      },
      {
        time: 1648490160000,
        latency: 0,
        ttl: 64,
      },
      {
        time: 1648490400000,
        latency: 0,
        ttl: 64,
      },
      {
        time: 1648490640000,
        latency: 0,
        ttl: 64,
      },
      {
        time: 1648490880000,
        latency: 0,
        ttl: 64,
      },
      {
        time: 1648491120000,
        latency: 0,
        ttl: 64,
      },
      {
        time: 1648491360000,
        latency: 0,
        ttl: 64,
      },
      {
        time: 1648491600000,
        latency: 0,
        ttl: 64,
      },
      {
        time: 1648491840000,
        latency: 0,
        ttl: 64,
      },
      {
        time: 1648500240000,
        latency: 0,
        ttl: 64,
      },
      {
        time: 1648500480000,
        latency: 0,
        ttl: 64,
      },
      {
        time: 1648500720000,
        latency: 0,
        ttl: 64,
      },
      {
        time: 1648500960000,
        latency: 0,
        ttl: 64,
      },
      {
        time: 1648501200000,
        latency: 0,
        ttl: 64,
      },
      {
        time: 1648501440000,
        latency: 0,
        ttl: 64,
      },
      {
        time: 1648501680000,
        latency: 0,
        ttl: 64,
      },
      {
        time: 1648501920000,
        latency: 0,
        ttl: 64,
      },
      {
        time: 1648502160000,
        latency: 0,
        ttl: 64,
      },
      {
        time: 1648502400000,
        latency: 0,
        ttl: 64,
      },
      {
        time: 1648502640000,
        latency: 0,
        ttl: 64,
      },
      {
        time: 1648502880000,
        latency: 0,
        ttl: 64,
      },
      {
        time: 1648503120000,
        latency: 0,
        ttl: 64,
      },
      {
        time: 1648503360000,
        latency: 0,
        ttl: 64,
      },
      {
        time: 1648503600000,
        latency: 0.25,
        ttl: 64,
      },
      {
        time: 1648559520000,
        latency: 0,
        ttl: 64,
      },
      {
        time: 1648559760000,
        latency: 0,
        ttl: 64,
      },
      {
        time: 1648560000000,
        latency: 0,
        ttl: 64,
      },
      {
        time: 1648560240000,
        latency: 0.25,
        ttl: 64,
      },
      {
        time: 1648560480000,
        latency: 0,
        ttl: 64,
      },
    ],
  },
];
export const getEnvironmentResponse = [
  false,
  null,
  {
    envName: "hto env 01",
    envId: 144,
    envUbiqubeId: "00pA144",
    description: "2nd",
    cloudVendor: "Cloud/AWS",
    orgName: "htoExtRef",
    envIp: "127.0.0.1",
    envHostname: "temp-host",
    username: "12",
    envReference: "0419674e-a030-4fb9-b47a-94bcb8de6d0e",
    importFlag: false,
    serviceId: "126087",
    serviceName: null,
    envWFUri:
      "Process/cloudclapp-wf/Provision_an_EKS_Cluster__AWS_/Provision_an_EKS_Cluster__AWS_",
    appWFUri:
      "Process/cloudclapp-wf/Provision_Apps_EKS_Cluster/Provision_Apps_EKS_Cluster",
    deploymentCount: 49,
    userCount: 4,
    scanData: null,
    cclapOwner: 125,
    envEntityId: 323,
  },
  {
    "cache-control": "no-cache, no-store, max-age=0, must-revalidate",
    connection: "close",
    "content-length": "576",
    "content-type": "application/json",
    date: "Wed, 23 Mar 2022 04:05:00 GMT",
    expires: "0",
    pragma: "no-cache",
    server: "nginx/1.21.6",
    vary: "Accept-Encoding",
    "x-content-type-options": "nosniff",
    "x-frame-options": "DENY",
    "x-powered-by": "Express",
    "x-xss-protection": "1; mode=block",
    status: 200,
  },
  null,
];

export const environmentSummary = {
  environmentCount: 6,
  usersCount: 3,
  managerList: [
    {
      id: 142,
      baseRole: {
        id: 3,
        name: "Privileged Manager",
      },
      externalReference: "00pG142",
      login: "htoOrgPrivMan",
      operator: {
        id: 29,
        abonnes: [],
        baseUrl: null,
        isPartner: false,
        name: "brand new hto org!",
        prefix: "00p",
        address: {
          city: "",
          country: "",
          fax: "",
          mail: "",
          phone: "",
          streetName1: "",
          streetName2: "",
          streetName3: "",
          zipCode: "",
        },
        customersCount: 0,
        adminsCount: 0,
        managersCount: 0,
        externalReference: "htoExtRef",
      },
      acteurId: 515,
      address: {
        city: "",
        country: "",
        fax: null,
        mail: "hto@ubiqube.com",
        phone: "",
        streetName1: "",
        streetName2: "",
        streetName3: "",
        zipCode: "",
      },
      delegationProfileId: 0,
      userType: 2,
      delegations: null,
      delegationsPerCustomer: {},
      netceloId: {
        ubiID: "00pG142",
        thePrefix: "00p",
        id: 142,
        customerPrefix: "00p",
      },
      isExternalAuth: false,
      activationKey: null,
      ccla: false,
      attachedCustomerIds: [143, 144, 145, 151, 149, 150],
      firstname: "",
      attachedOperatorIds: null,
      manageAllUsers: true,
      name: "hto org priv man",
      sort: "NAME",
      ldapAuthentication: false,
      delegationProfilePerCustomer: {},
    },
    {
      id: 140,
      baseRole: {
        id: 4,
        name: "Manager",
      },
      externalReference: "00pG140",
      login: "htoOrgMan",
      operator: {
        id: 29,
        abonnes: [],
        baseUrl: null,
        isPartner: false,
        name: "brand new hto org!",
        prefix: "00p",
        address: {
          city: "",
          country: "",
          fax: "",
          mail: "",
          phone: "",
          streetName1: "",
          streetName2: "",
          streetName3: "",
          zipCode: "",
        },
        customersCount: 0,
        adminsCount: 0,
        managersCount: 0,
        externalReference: "htoExtRef",
      },
      acteurId: 511,
      address: {
        city: "",
        country: "",
        fax: null,
        mail: "hto@ubiqube.com",
        phone: "",
        streetName1: "",
        streetName2: "",
        streetName3: "",
        zipCode: "",
      },
      delegationProfileId: 0,
      userType: 2,
      delegations: null,
      delegationsPerCustomer: {},
      netceloId: {
        ubiID: "00pG140",
        thePrefix: "00p",
        id: 140,
        customerPrefix: "00p",
      },
      isExternalAuth: false,
      activationKey: null,
      ccla: false,
      attachedCustomerIds: [144],
      firstname: "",
      attachedOperatorIds: null,
      manageAllUsers: false,
      name: "hto org dev guy",
      sort: "NAME",
      ldapAuthentication: false,
      delegationProfilePerCustomer: {},
    },
    {
      id: 125,
      baseRole: {
        id: 2,
        name: "Administrator",
      },
      externalReference: "NCLG125",
      login: "hto@openmsa.com",
      operator: {
        id: 29,
        abonnes: [],
        baseUrl: null,
        isPartner: false,
        name: "brand new hto org!",
        prefix: "00p",
        address: {
          city: "",
          country: "",
          fax: "",
          mail: "",
          phone: "",
          streetName1: "",
          streetName2: "",
          streetName3: "",
          zipCode: "",
        },
        customersCount: 0,
        adminsCount: 0,
        managersCount: 0,
        externalReference: "htoExtRef",
      },
      acteurId: 454,
      address: {
        city: "",
        country: "",
        fax: null,
        mail: "hto@openmsa.com",
        phone: "",
        streetName1: "",
        streetName2: "",
        streetName3: "",
        zipCode: "",
      },
      delegationProfileId: 0,
      userType: 2,
      delegations: null,
      delegationsPerCustomer: {},
      netceloId: {
        ubiID: "00pG125",
        thePrefix: "00p",
        id: 125,
        customerPrefix: "00p",
      },
      isExternalAuth: false,
      activationKey: null,
      ccla: false,
      attachedCustomerIds: null,
      firstname: "",
      attachedOperatorIds: [29],
      manageAllUsers: false,
      name: "hto@openmsa.com",
      sort: "NAME",
      ldapAuthentication: false,
      delegationProfilePerCustomer: {},
    },
  ],
};


export const permissionProfiles = {
  environments: {
    general: {
      create: {
        name: 'create',
        access: false
      },
      tearDown: {
        name: 'tearDown',
        access: true
      },
      delete: {
        name: 'delete',
        access: false
      },
      access: true
    },

  },
  deployments: {
    general: {
      create: {
        name: 'create',
        access: true
      },
      action: {
        name: 'action',
        access: false
      },
      access: true
    },

  },
  applications: {
    dockerHub: {
      view: {
        name: 'view',
        access: true
      },
      access: true
    },
    virtualMachine: {
      view: {
        name: 'view',
        access: false
      },
      access: false
    },
    access: true
  },

  insights: {
    cost: {
      view: {
        name: 'view',
        access: true
      },
      access: true
    },
  },

  governance: {
    auditLogs: {
      view: {
        name: 'view',
        access: false
      },
      access: false
    },
  },
}
