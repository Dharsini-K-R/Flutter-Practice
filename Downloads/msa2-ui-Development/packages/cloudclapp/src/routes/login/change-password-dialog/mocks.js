export const linkValidationResponse = {
  message: "Link is valid",
};
