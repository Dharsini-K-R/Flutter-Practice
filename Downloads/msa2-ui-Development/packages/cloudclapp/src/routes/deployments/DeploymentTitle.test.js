import React from "react";
import { renderWithProvider } from "cloudclapp/src/utils/test-utils";
import DeploymentTitle from "./DeploymentTitle";
import { environment } from "./mocks";

import initialState from "cloudclapp/src/store/initialState";

const mockState = {
  ...initialState,
};

const id = 126;
const title = "Development";
const status = "OK";

describe("DeploymentTitle component", () => {
  it("renders correctly", () => {
    const { container, getByText } = renderWithProvider({
      childComponent: (
        <DeploymentTitle
          id={id}
          title={title}
          status={status}
          environment={environment}
        />
      ),
      mockState,
    });

    expect(container.firstChild).toMatchSnapshot();

    expect(getByText("Development")).toBeTruthy();
  });
});
