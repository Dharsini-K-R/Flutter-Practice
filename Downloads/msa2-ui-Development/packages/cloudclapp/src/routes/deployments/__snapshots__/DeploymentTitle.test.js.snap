// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeploymentTitle component renders correctly 1`] = `
<div
  class="makeStyles-boxWrapper"
>
  <div
    class="MuiGrid-root makeStyles-boxPadding MuiGrid-container"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-true"
    >
      <div
        class="makeStyles-titleContainer"
      >
        <div
          class="MuiGrid-root MuiGrid-item"
        >
          <span
            class="MuiBadge-root"
          >
            <i
              aria-hidden="true"
              class="la la-cloud makeStyles-icon"
              role="presentation"
              style="font-size: 28px;"
            />
            <span
              class="MuiBadge-badge makeStyles-badge makeStyles-badge MuiBadge-anchorOriginBottomRightCircular MuiBadge-dot"
            />
          </span>
          <span
            class="makeStyles-label"
          />
        </div>
        <div
          class="MuiGrid-root MuiGrid-item"
        >
          <p
            class="MuiTypography-root makeStyles-text MuiTypography-body1"
            id="DEPLOYMENTS_DEPLOYMENT_TITLE_126"
          >
            Development
          </p>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-justify-content-xs-flex-end MuiGrid-grid-xs-6"
      style="align-items: center;"
      title=""
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text"
        id="DEPLOYMENTS_ADD_DEPLOYMENT_BUTTON_126"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          <div
            class="MuiGrid-root MuiGrid-item"
          >
            <i
              aria-hidden="true"
              class="la la-plus-circle makeStyles-addIcon"
              role="presentation"
            />
          </div>
          <div
            class="MuiGrid-root MuiGrid-item"
          >
            <p
              class="MuiTypography-root makeStyles-buttonText MuiTypography-body1"
              id="DEPLOYMENTS_ADD_DEPLOYMENT_BUTTON_126"
            >
              Add Deployment
            </p>
          </div>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;
